decompressed/base/etc/init.d/rootdevice

```diff
@@ -249,59 +198,114 @@ install_gui() {
     rm /root/.reapply_due_to_upgrade
   fi
 
-  echo "  Process done. Your modem is now rooted!" >/dev/console
+  logger_command "Process complete. tch-nginx-gui is now installed!"
+  echo "  Process complete. tch-nginx-gui is now installed!" >/dev/console
 }
 
-check_bank() {
-  bank_check=$(uci get -q modgui.var.bank_check)
+is_dual_bank() {
+  grep -c "\"bank_2\"" /proc/mtd >/dev/null
+}
+
+get_bank_mtd() {
+  cat /proc/mtd  | grep \"$1\" | sed 's/:.*//'
+}
+
+check_and_apply_obp() {
+  check_obp=$(uci get -q modgui.var.check_obp)
   bootedbank=$(cat /proc/banktable/booted)
   activebank=$(cat /proc/banktable/active)
-  env_config_bank1="/overlay/bank_1/etc/config/modgui"
-  if [ "$bootedbank" != "$activebank" ]; then
-    echo "  Detected corrupted active bank... Restoring..." >/dev/console
-    restore_bank
-  fi
-
-  if [ "$bootedbank" == "bank_2" ]; then
-    if [ -f $env_config_bank1 ]; then
-      get_config_bank1=$(< $env_config_bank1 grep bank_check | awk '{print $3}' | sed "s/'//g")
-      if [ -n "$get_config_bank1" ] && [ ! "$(uci -q get modgui.var.bank_check)" ]; then
-        uci set modgui.var.bank_check="$get_config_bank1"
+  notbooted=$(cat /proc/banktable/notbooted)
+  notbooted_modgui_configfile="/overlay/$notbooted/etc/config/modgui"
+
+  if [[ "$bootedbank" != "$activebank" && "$bootedbank" == "bank_2" ]]; then
+    #TODO: add checks to be sure the bank_1 is unbootable (is this really needed?)
+    #TODO: add checks to be sure the bank_1 overlay is SAFE and bootable (empty/prerooted?)
```

Comment:
@LuKePicci su questi due TODO che mi dici? pensi sia necessario il primo check? non credo ci sia modo di arrivarci in quella condizione se non hai un bank_1 corrotto (forse failboot volontario?)

per il secondo le scelte penso sono o 
`[ -d "/overlay/bank_1" ] && rm -rf "/overlay/bank_1"`
o assicurarci che il contenuto di "/overlay/bank_1" sia == pacchetto base

Reply:
Ho più idee che soluzioni.

La prima si può anche determinare se trovi il modo di verificarne la firma. Un failure sulla verifica della firma implica la non avviabilità. Però comunque non intraprenderei nessuna azione oltre che segnalare la cosa.

Per la seconda, in teoria l'overlay del primo banco è da considerarsi sano se è vuoto e la partizione non è piena. Se poi vuoi forzare quell'overlay in una situazione artificialmente safe puoi provare a mettergli in uci-defaults uno script che se trova la partizione piena fa l'erase, altrimenti imposta l'handler del tasto reset su mtd erase, attende una ventina di secondi e ritorna exit 0, verrebbe eseguito da /etc/init.d/boot, quindi abbastanza presto, e ritarderebbe abbastanza tutto il resto da permettere il reset. L'exit 0 serve ad evitare che venga cancellato.

Reply:
Direi andati anche questi, per ora consideriamo safe un overlay vuoto.

il semplice log se il bank1 è valido dovrebbe bastare fatto in questo modo con il check passiveversion